src/nvim/macros.h

```diff
@@ -120,8 +120,10 @@
 /* mch_open_rw(): invoke os_open() with third argument for user R/W. */
 #if defined(UNIX)  /* open in rw------- mode */
 # define mch_open_rw(n, f)      os_open((n), (f), (mode_t)0600)
+#elif defined(WIN32)
+# define mch_open_rw(n, f)      os_open((n), (f), S_IREAD | S_IWRITE)
```

Comment:
@erw7 I included S_IREAD here, if only to be explicit. That matches the `0600` permissions for the UNIX case, and it also matches what Vim does.